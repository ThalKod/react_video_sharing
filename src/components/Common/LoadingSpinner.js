import React from "react";
import { BeatLoader } from "react-spinners";

export default ({text, medium}) => (
    <div className="content-wrapper">
      <div className="container">
        <div className="row">
          <div className={medium? "col-lg-6 upload-page": "col-lg-12 upload-page" }>
            <div className="u-area">
              <BeatLoader
                  sizeUnit="px"
                  size={30}
                  color='#123abc'
                  loading
              />
              {text && <p>{text}</p>}
            </div>
          </div>
        </div>
      </div>
    </div>
);
