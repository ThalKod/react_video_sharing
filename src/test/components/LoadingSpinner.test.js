import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import LoadingSpinner from "components/Common/LoadingSpinner";

describe("LoadingSpinner", () => {
  it("should correctly render the loading spinner", () => {
    const wrapper = shallow(<LoadingSpinner text="Loading"/>);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.contains(<p>Loading</p>)).toBe(true);
  })
});
