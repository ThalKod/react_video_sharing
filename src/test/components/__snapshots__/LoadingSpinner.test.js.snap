// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner should correctly render the loading spinner 1`] = `
<div
  className="content-wrapper"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-12 upload-page"
      >
        <div
          className="u-area"
        >
          <onlyUpdateForKeys(Loader)
            color="#123abc"
            css=""
            loading={true}
            margin="2px"
            size={30}
            sizeUnit="px"
          />
          <p>
            Loading
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
